lib/realm-verifier/src/lib.rs

```diff
@@ -7,55 +7,99 @@ use tinyvec::ArrayVec;
 
 pub const MAX_MEASUREMENT_SIZE: usize = 64;
 
-pub type MeasurementValue = ArrayVec<[u8; MAX_MEASUREMENT_SIZE]>;
+#[derive(Debug, Clone, Copy)]
+pub struct MeasurementValue {
+    value: ArrayVec<[u8; MAX_MEASUREMENT_SIZE]>,
+}
+
+impl MeasurementValue {
+    pub fn init(len: usize) -> Self {
+        let mut av = ArrayVec::new();
+        av.resize(len, 0);
+        Self {
+            value: av
+        }
+    }
+
+    pub fn as_slice(&self) -> &[u8] {
+        self.value.as_slice()
+    }
+
+    pub fn as_mut_slice(&mut self) -> &mut[u8] {
+        self.value.as_mut_slice()
+    }
+}
+
+#[derive(Debug)]
+pub struct HashAlgo {
```

Comment:
This struct should be enum. In its current form you can create inconsistent values.

Reply:
Not enum, but fixed.